app/Http/Controllers/ScheduleController.php

```diff
@@ -148,184 +149,94 @@ public function destroy($id)
-        }
-        else
-        {
-            $newId = "";
-            $newName = "";
-        }
-        
-        // append current change
-        array_push($revisions, ["entry id"    => $entry->id,
-                                "job type"    => $entry->getJobType->jbtyp_title,
-                                "action"      => $action,
-                                "old id"      => $oldId,
-                                "old value"   => $oldName,
-                                "old comment" => $oldComment,
-                                "new id"      => $newId,
-                                "new value"   => $newName,
-                                "new comment" => $newComment,
-                                "user id"     => Session::get('userId') != NULL ? Session::get('userId') : "",
-                                "user name"   => Session::get('userId') != NULL ? Session::get('userName') . ' (' . Session::get('userClub') . ')' : "Gast",
-                                "from ip"     => Request::getClientIp(),
-                                "timestamp"   => (new DateTime)->format('d.m.Y H:i:s') ]
-                    );      
-
-        // encode and save
-        $schedule->entry_revisions = json_encode($revisions);
-                        
-        $schedule->save();
-    }
-
-
     /**
     * Create all new scheduleEntries with entered information.
     *
     * @return Collection scheduleEntries
     */
-    public static function createScheduleEntries()
+    public static function createShifts($scheduleId, $isNewEvent = true)
     {
-        $scheduleEntries = new Collection;
-
-        // parsing jobtype entries
-        for ($i=1; $i <= Input::get("counter"); $i++) {
-
-            // skip empty fields
-            if (!empty(Input::get("jbtyp_title" . $i))) 
-            {       
-                // check if job type exists
-                $jobType = Jobtype::where('jbtyp_title', '=', Input::get("jbtyp_title" . $i))
-                                  ->where('jbtyp_time_start', '=', Input::get("jbtyp_time_start" . $i))
-                                  ->where('jbtyp_time_end', '=', Input::get("jbtyp_time_end" . $i))
-                                  ->first();
+        $inputShifts = Input::get("shifts");
+        $amount = count($inputShifts["title"]);
+
+        Shift::where('schedule_id', $scheduleId)
+            ->whereNotIn('id', $inputShifts["id"])
+            ->get()
+            ->each(function(Shift $shift) {
+                Logging::shiftDeleted($shift);
+                $shift->delete();
+            });
+
+        for ($i = 0; $i < $amount; ++$i) {
+            if ($inputShifts["title"][$i] !== "") {
+                // Cloned shifts have the "id" field set to "", so we will create a new model in this case
+                $shift = Shift::firstOrNew(["id" => $inputShifts["id"][$i]]);
+
+                // If there was a shifttype passed and one with the correct title exists, use this one
```

Comment:
I think there is still some ambiguity around this issue @4D44H. 

Suppose that I cloned a shift with an existing shift type, and I change the time for this shift. Should a new shifttype be created? If so, the `start` and `end` attributes for shifts are redundant (because they will always coincide with the shifttype). 
If no new shifttype should be created, we do not have to check for start/end time here. (Btw: this is not the place to check for time, its actually some lines down)